src/util/actor.js

```diff
@@ -107,20 +108,22 @@ class Actor {
                 cancel.cancel();
             }
         } else {
-            if (isWorker() || data.mustQueue) {
-                // In workers, store the tasks that we need to process before actually processing them. This
-                // is necessary because we want to keep receiving messages, and in particular,
-                // <cancel> messages. Some tasks may take a while in the worker thread, so before
-                // executing the next task in our queue, postMessage preempts this and <cancel>
-                // messages can be processed. We're using a MessageChannel object to get throttle the
-                // process() flow to one at a time.
+            if (isWorker() && data.mustQueue) {
```

Comment:
We need to pass the responses from getImages and getGlyphs to the scheduler. Checking for the presence of `callback.metadata` in actor.js might be enough to decide whether to do that but I think letting the scheduler decide that might be slightly cleaner

The queuing on the main thread was [intentional](https://github.com/mapbox/mapbox-gl-js/pull/9031) but it looks like it might not be needed since we dropped IE. I think [this]((https://github.com/mapbox/mapbox-gl-js/blob/719301fcae7f359453cdfc0bf699a40734270cae/src/util/ajax.js#L245-L246)) is the only case where we did queuing on the main thread. It was also applied to iOS Safari < 12.1 but I don't think it was actually  needed there... not sure though.  @arindam1993 do you remember if the queuing was only needed for IE?

Reply:
Also old Safari verions wherein `AbortController` doesn't actually abort fetches.